review/npm.md

```diff
@@ -244,19 +284,21 @@ run.
         1. To run tests, run [`npm
            install-ci-test`](https://docs.npmjs.com/cli/v8/commands/npm-install-ci-test).
 
-    1. If you need to run a CLI package from the registry, ensure the package is a part of
-       the dependencies defined in your project via the `package.json` file, prior to being installed at build-time in your CI or otherwise automated environment.
-
-1. If a project is a CLI application (`main` entry in the manifest file),
-   developers may publish a shrinkwrap.json.
-
-1. If a project is a library (`lib` entry in the manifest file), a
-   shrinkwrap.json should ***not*** be published. The reasoning is that version
-   resolution should be left to the package consumer. Allow all versions from
-   the minimum to the latest you support, e.g., `^m.n.o` to pin to a major
-   range; `~m.n.o` to pin to a minor range. Avoid versions with critical
-   vulnerabilities as much as possible. Visit the [semver
-   calculator](https://semver.npmjs.com/) to help you define the ranges.
+    1. If you need to run a standalone CLI package from the registry, ensure the package is a part of
+       the dependencies defined in your project via the `package.json` file, prior to
+       being installed at build-time in your CI or otherwise automated environment.
+
+1. In **non-privileged environments**, maintainers may **ignore** the lockfile. This is particularly useful in
+   situations where they want to exercise a wide range of dependency versions in order to discover / fix problems before
+   their users do. This is useful for maintainers of libraries and standalone CLI projects
+   without a shrinkwrap.json. The reasoning is that many downstream users will use `npm install` to install a dependency,
+   so using floating versions in (non-privileged) tests can be beneficial.
+   
+   1. If you run CI via GitHub Actions, a non-privileged environment is a workflow **without** access to GitHub secrets and with its
+   permissions defined as `permissions: read-all` at the top of the workflow. 
```

Comment:
i think there's a lot of other permissions values besides "read-all" where this may be valid? eg https://github.com/nvm-sh/nvm/blob/master/.github/workflows/latest-npm.yml#L27, https://github.com/nvm-sh/nvm/blob/6cc90a4b8daa44ad3050828c2bf98fa77a16bf90/.github/workflows/shellcheck.yml#L47, etc.

Reply:
Done, PTAL